itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestStatsReplicationScenarios.java

```diff
@@ -269,11 +294,23 @@ private String dumpLoadVerify(List<String> tableNames, String lastReplicationId,
     WarehouseInstance.Tuple dumpTuple = primary.run("use " + primaryDbName)
             .dump(primaryDbName, lastReplicationId, withClauseList);
 
+
     // Load, if necessary changing configuration.
     if (parallelLoad) {
       replica.hiveConf.setBoolVar(HiveConf.ConfVars.EXECPARALLEL, true);
     }
 
+    // Fail load if for testing failure and retry scenario. Fail the load while setting
+    // checkpoint for a table in the middle of list of tables.
+    if (failRetry) {
+      if (lastReplicationId == null) {
+        failBootstrapLoad(dumpTuple, tableNames.size()/2);
+      } else {
+        failIncrementalLoad(dumpTuple, tableNames.size()/2);
```

Comment:
It is not directly mapped that one event per table. So, this value of tableNames.size()/2 may fail even before applying update state event. If we want to fail the incremental load after a fixed event, then need to get the event count by dumping it after that operation at source.

Reply:
We are counting UpdateTableStats or UpdatePartStats events and not every event. So, we will fail only after encountering no of tables/2 events of those types. So it can not fail before applying update stats events. But to be on the safer side, I have changed the code to fail after second event so that we have at least one successful application before we fail. Since we are performing multiple insert events per table, we can be sure that there are at least 2 events of each type.